src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/INameSpaceSymbolExtensions.cs

```diff
@@ -1,20 +1,28 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Collections.Generic;
 using Microsoft.CodeAnalysis;
 
 namespace Azure.ClientSdk.Analyzers
 {
     public static class INamespaceSymbolExtensions
     {
-        public static string GetFullNamespaceName(this INamespaceSymbol namespaceSymbo)
+        public static string GetFullNamespaceName(this INamespaceSymbol namespaceSymbol)
         {
-            if (namespaceSymbo is { ContainingNamespace: not null and { IsGlobalNamespace: false} })
+            return string.Join(".", namespaceSymbol.GetAllNamespaces());
```

Comment:
Perhaps we should re-think this implementation a bit. We are building a list that will end up in a `String.Join`, should we consider a StringBuilder? We can also avoid the list `Reverse` of the list by appending at position 0 of the builder.

Also, given that the [consumer](https://github.com/Azure/azure-sdk-tools/pull/6916/files/2e313f401eeb8adc5f3d4c7c1ec18eaace8af0be..9abf0865f130aee50f3827567e7b5e1845a962e4#diff-b7bcf510670ac60c111caaae450c5b661b6e32cd4f75f8f55f42e2b8c1416d71R39) of this turns the string back to a stringBuilder, perhaps this should just return a string builder?